### ISSUE TITLE
Order notes helper function

### ISSUE THREAD

**Issue by claudiulodro** [woocommerce/woocommerce#15982]
This was first reported on [the forums](https://wordpress.org/support/topic/get_comments-function-is-not-returning-the-order-notes-in-wc-3-1-0/), but I have reproduced the problem and confirmed get_comments worked with orders in 3.0.8.

In 3.0.8 `get_comments( array( 'post_id' => $order->get_id() ) )` returns all of the order notes for an order.

In 3.1/master, the same thing returns an empty array for me and returns product reviews for the user in the forums.

EDIT: This is now a ticket for a helper function to make it easy for devs to get order notes (e.g. `wc_get_order_notes`).

**Comment by claudiosanches** [comment]
I think this is a bug on 3.0.8, since the correct is to be impossible to get any results from `get_comments()` because this code: https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-comments.php#L70-L84

And for this reason every time that we try query order notes we use something like:

https://github.com/woocommerce/woocommerce/search?utf8=%E2%9C%93&q=remove_filter%28+%27comments_clauses%27%2C+array%28+%27WC_Comments%27%2C+%27exclude_order_comments%27+%29%2C+10%2C+1+%29%3B&type=

https://github.com/woocommerce/woocommerce/blob/99bc154400c6239de4d3e07a1cce8f6593566091/includes/admin/meta-boxes/class-wc-meta-box-order-notes.php#L28-L40

This sounds like bug as a feature in 3.0 and finally fixed in 3.1.

**Comment by claudiosanches** [comment]
Best solution here is to have a helper to get order notes, and avoid this kind of confusion, even more now that we have plans to move to a custom table.

**Comment by claudiulodro** [comment]
I think it's "breaking" now because we lost this line:
https://github.com/woocommerce/woocommerce/commit/b32d5802b74eff376e97b0fc7409e6576303c8e3#diff-f59ccabc5e71b1276fd9ec1e15119b5eL70

It allows order notes in get_comments in the admin area.

I agree, a wc_get_order_notes or something like that would be the best solution.

**Comment by claudiosanches** [comment]
This changes are necessary to fix #15509 and still is required to properly remove the filters before query comments.
Let's move it to 3.2, should be very easy to create an helper for it, on forum we can just reply saying that the correct way to query is with:

```php
$args = array(
	'status' => 'approve',
	'post_id' => $order->get_id(),
);

remove_filter( 'comments_clauses', array( 'WC_Comments', 'exclude_order_comments' ), 10, 1 );
$comments = get_comments( $args );
add_filter( 'comments_clauses', array( 'WC_Comments', 'exclude_order_comments' ), 10, 1 );
```
